core/src/test/java/com/novoda/merlin/service/ConnectivityChangesForwarderTest.java

```diff
@@ -110,6 +110,30 @@ public void givenFetchingWithPing_whenUnsuccessful_thenCallsOnDisconnect() {
     }
 
     @Test
+    public void givenNetworkWasConnected_butNullBindListener_whenNotifyingOfInitialState_thenForwardsNetworkAvailableToListener() {
+        connectivityChangesForwarder = new ConnectivityChangesForwarder(
+                networkStatusRetriever, disconnectCallbackManager, connectCallbackManager, null, endpointPinger
+        );
+        givenNetworkWas(CONNECTED);
+
+        connectivityChangesForwarder.forwardInitialNetworkStatus();
+
+        verify(bindCallbackManager, never()).onMerlinBind(any(NetworkStatus.class));
+    }
+
+    @Test
+    public void givenEndpointIsUnavailable_andNullBindListener_whenNotifyingOfInitialState_thenForwardsNetworkAvailableToListener() {
+        connectivityChangesForwarder = new ConnectivityChangesForwarder(
+                networkStatusRetriever, disconnectCallbackManager, connectCallbackManager, null, endpointPinger
+        );
+        given(networkStatusRetriever.retrieveNetworkStatus()).willReturn(AVAILABLE_NETWORK);
```

Comment:
@joetimmins I've tried to make it a little bit more obvious in 7d2c617 but it is kind of driven by whether or not a `forward` event has already occurred.